docs/content/docs/recipes/audience-react.md

```diff
@@ -0,0 +1,396 @@
+
+```js
+  if (!fluidMembers || !currentMember) return (<div/>);
+
+  return (
+      <div>
+        <AudienceList fluidMembers={fluidMembers} currentMember={currentMember}/>
+      </div>
+  )
+```
+
+{{< callout note >}}
+
+Connection transitions can result in short timing windows where `getMyself` returns `undefined`. This is because the current client connection will not have been added to the audience yet, so a matching connection ID cannot be found. To prevent React from rendering a page with no audience members, we add a listener to call `updateMembers` on `membersChanged`. This works since the service audience emits a `membersChanged` event when the container is connected.
+
+{{< /callout >}}
+
+### Create the view
+
+1. Replace `TODO 8` with the following code. Note we are rendering a list component for each member passed from the `AudienceDisplay` component. For each member, we first compare `member.userId` to `currentMember.userId` to check if that member `isSelf`. This way, we can differentiate the client user from the other users and display the component with a different color. We then push the list component to a `list` array. Each component will display member data such as `userId` `userName` and `additionalDetails`.
+
+```js
+  const currentMember = data.currentMember;
+  const fluidMembers = data.fluidMembers;
+
+  const list = [];
+  fluidMembers.forEach((member, key) => {
+      const isSelf = (member.userId === currentMember.userId);
+      const outlineColor = isSelf ? 'blue' : 'black';
+
+      list.push(
+        <div style={{
+          padding: '1rem',
+          margin: '1rem',
+          display: 'flex',
+          outline: 'solid',
+          flexDirection: 'column',
+          maxWidth: '25%',
+          outlineColor
+        }} key={key}>
+          <div style={{fontWeight: 'bold'}}>Name</div>
+          <div>
+              {member.userName}
+          </div>
+          <div style={{fontWeight: 'bold'}}>ID</div>
+          <div>
+              {member.userId}
+          </div>
+          <div style={{fontWeight: 'bold'}}>Connections</div>
+          {
+              member.connections.map((data, key) => {
+                  return (<div key={key}>{data.id}</div>);
+              })
+          }
+          <div style={{fontWeight: 'bold'}}>Additional Details</div>
+          { JSON.stringify(member.additionalDetails, null, '\t') }
+        </div>
+      );
+  });
+```
+
+1. Replace `TODO 9` with the following code.
+
+```js
+  return (
+      <div>
+          {list}
+      </div>
+  );
+```
+
+### Setup SelectUser component
+
+1. Create and open a file `\src\SelectUser.js` in your code editor. Add the following `import` statements and functional components:
+
+```js
+import { useState } from 'react';
+
+export const SelectUser = (props) => {
```

Comment:
should this be named selectUser (camel case) as JS functions normally are?

Reply:
This is also a function component so it should probably be renamed to a noun.

Reply:
Got it. Thanks.